src/pwncore/routes/ctf/__init__.py

```diff
@@ -10,19 +24,89 @@
 
 router = APIRouter(prefix="/ctf", tags=["ctf"])
 router.include_router(start_router)
+
 # Routes that do not need a separate submodule for themselves
 
 
+# For testing purposes. Replace function with POST method
+def get_flag():
+    return "pwncore{flag_1}"
+
+
 @router.get("/list")
 async def ctf_list():
-    # Get list of ctfs
-    return [
-        {"name": "Password Juggling", "ctf_id": 2243},
-        {"name": "hexane", "ctf_id": 2242},
-    ]
+    problems = await Problem_Pydantic.from_queryset(Problem.all())
+    return problems
+
+
+# For testing purposes only. flag to be passed in body of POST function.
+@router.get("/{ctf_id}/flag")
+async def flag_get(ctf_id: int, response: Response):
+    problem = await Problem.exists(id=ctf_id)
+    if not problem:
+        response.status_code = 404
+        return {"msg_code": config.msg_codes["ctf_not_found"]}
+
+    status = await SolvedProblem.get_or_none(team_id=get_team_id(), problem_id=ctf_id)
+    if status:
+        response.status_code = 401
+        return {"msg_code": config.msg_codes["ctf_solved"]}
+
+    check_solved = await Container.get_or_none(
+        team_id=get_team_id(), flag=get_flag(), problem_id=ctf_id
+    )
+    if check_solved:
+        await SolvedProblem.create(team_id=get_team_id(), problem_id=ctf_id)
+        return {"status": True}
+    return {"status": False}
+
+
+@router.get("/{ctf_id}/hint")
+async def hint_get(ctf_id: int, response: Response):
+    problem = await Problem.exists(id=ctf_id)
+    if not problem:
+        response.status_code = 404
+        return {"msg_code": config.msg_codes["ctf_not_found"]}
+
+    viewed_hints = (
+        await Hint.filter(problem_id=ctf_id, viewedhints__team_id=get_team_id())
+        .order_by("-order")
+        .first()
+    )
+    if viewed_hints:
+        if not await Hint.exists(problem_id=ctf_id, order=viewed_hints.order + 1):
+            response.status_code = 403
+            return {"msg_code": config.msg_codes["hint_limit_reached"]}
+
+        hint = await Hint.get(problem_id=ctf_id, order=viewed_hints.order + 1)
+
+    else:
+        hint = await Hint.get(problem_id=ctf_id, order=0)
+
+    await ViewedHint.create(hint_id=hint.id, team_id=get_team_id())
+    return {"text": hint.text, "order": hint.order}
+
+
+@router.get("/{ctf_id}/viewed_hints")
+async def viewed_problem_hints_get(ctf_id: int):
+    viewed_hints = await Hint_Pydantic.from_queryset(
+        Hint.filter(problem_id=ctf_id, viewedhints__team_id=get_team_id())
```

Comment:
`viewedhints__team_id` -> `viewedhint__teami_id`

Reply:
or you can create and explicit relation and set `related_name` to whatever you want

Reply:
doesn't need to changed either, this is at your discretion